src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/ActiveWorkspaceProjectContextTracker.cs

```diff
@@ -83,5 +89,24 @@ public void UnregisterContext(IWorkspaceProjectContext context)
             if (!changed)
                 throw new InvalidOperationException("'context' has not been registered or has already been unregistered");
         }
+
+        private string GetWorkspaceContextIdFromActiveContext()
+        {
+
+            return _threadingService.ExecuteSynchronously(() =>
+            {
+                try
+                {
+                    // If we're never been set an active context, we just
+                    // pick one based on the active configuration.
+                    return _activeWorkspaceProjectContextHost.OpenContextForWriteAsync(a => Task.FromResult(a.ContextId));
```

Comment:
This needs `await` in order to observe the exception, right? Or is `OperationCanceledException` special?

Reply:
> Or is OperationCanceledException special

Nope, good catch.

Reply:
```suggestion
                    return await _activeWorkspaceProjectContextHost.OpenContextForWriteAsync(a => Task.FromResult(a.ContextId));
```